src/components/ContinuationIn/ExtraproRegistration.vue

```diff
@@ -0,0 +1,331 @@
+  }
+})
+export default class ExtraproRegistration extends Vue {
+  @Getter(useStore) getContinuationInBusinessInfo!: any
+  @Getter(useStore) getShowErrors!: boolean
+
+  @Action(useStore) setContinuationInBusinessInfo!: (x: any) => void
+
+  // Local properties
+  active = false
+  business = null // FUTURE: update BusinessLookupResultIF with new properties
+  checkbox = false
+
+  readonly checkboxLabel = 'I understand and acknowledge that the extraprovincial registration of this' +
+    ' business will be cancelled and made historical in the B.C. Registry once the continuation in' +
+    ' application has been submitted.'
+
+  /** Whether we have a looked-up business. */
+  get haveLookupBusiness (): boolean {
+    return !isEqual(this.business, EmptyBusinessLookup)
+  }
+
+  /** The identifier. */
+  get identifier (): string {
+    return this.business.identifier
+  }
+
+  /** The jurisdiction. */
+  get jurisdiction (): string {
+    const hj = this.business.homeJurisdiction
+    if (hj?.country) {
+      const country = getName(hj.country)
+      const region = (hj.region === 'FEDERAL' ? 'Federal' : hj.region)
+      if (region) return `${region}, ${country}`
+      return country
+    }
+    return null
+  }
+
+  /** The business name. */
+  get businessName (): string {
+    return this.business.name
+  }
+
+  /** The business number. */
+  get businessNumber (): string {
+    return this.business.bn
+  }
+
+  /** The incorporation date. */
+  get incorporationDate (): string {
+    // FUTURE: format date if needed
+    return this.business.registrationDate
+  }
+
+  /** Whether business is Active (otherwise it's Historical). */
+  get isBusinessActive (): boolean {
+    return (this.business.status === 'ACTIVE')
+  }
+
+  mounted (): void {
+    // get the business info object from the store or initialize it
+    this.business = this.getContinuationInBusinessInfo || { ...EmptyBusinessLookup }
+
+    // if mode is LOOKUP, set this component to active (which hides the other component)
+    if (this.business.mode === 'LOOKUP') this.active = true
+  }
+
+  setBusiness (business: BusinessLookupResultIF) {
+    this.business = { ...business } // for reactivity
+    this.business.mode = 'LOOKUP'
+    this.setContinuationInBusinessInfo(this.business)
+    this.active = true
+  }
+
+  /** Resets this component back to its initial state. */
+  reset () {
+    this.business = { ...EmptyBusinessLookup }
+    this.setContinuationInBusinessInfo(this.business)
+    // set this component to inactive (which shows the other component)
+    this.active = false
+    this.checkbox = false
+  }
+
+  // *** TODO: add watchers for jurisdiction and affidavit
+  @Watch('business')
+  @Watch('checkbox')
+  @Emit('valid')
+  private onComponentValid (): boolean {
+    // this component is active if we have an active business
+    return (this.haveLookupBusiness && this.isBusinessActive && this.checkbox)
+  }
+
+  @Watch('active', { immediate: true })
```

Comment:
This forces the watcher here to be executed immediately on load right?

Reply:
I think this is a neat way for components to "check in" and report their initial state to the parent (so the parent doesn't have to assume anything).

Reply:
Makes sense, OK. Thanks Sev!